### PR TITLE
2023.1 letsencrypt

### PR DESCRIPTION
**Pull request by mnasiadka**

### CODE REVIEW

File: ansible/roles/letsencrypt/tasks/pull.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (markgoddard) [inline ansible/roles/letsencrypt/tasks/pull.yml:1]: Should use service-images-pull, but should be fixed upstream.

Comment (markgoddard) [reply ansible/roles/letsencrypt/tasks/pull.yml:1]: https://review.opendev.org/c/openstack/kolla-ansible/+/904587

File: ansible/roles/loadbalancer/defaults/main.yml [hunk 2]
```diff
@@ -105,6 +134,13 @@ keepalived_default_volumes:
   - "/lib/modules:/lib/modules:ro"
   - "{{ 'haproxy_socket:/var/lib/kolla/haproxy/' if enable_haproxy | bool else '' }}"
   - "{{ 'proxysql_socket:/var/lib/kolla/proxysql/' if enable_proxysql | bool else '' }}"
+haproxy_ssh_default_volumes:
+  - "{{ node_config_directory }}/haproxy-ssh/:{{ container_config_directory }}/:ro"
+  - "/etc/localtime:/etc/localtime:ro"
+  - "{{ '/etc/timezone:/etc/timezone:ro' if ansible_facts.os_family == 'Debian' else '' }}"
+  - "haproxy_socket:/var/lib/kolla/haproxy/"
+  - "{{ 'letsencrypt:/etc/letsencrypt' if enable_letsencrypt | bool else omit }}"
+  - "{{ 'letsencrypt_certificates:/etc/haproxy/certificates' if enable_letsencrypt | bool else omit }}"
```
Comment (markgoddard) [inline ansible/roles/loadbalancer/defaults/main.yml:143]: Might be worth checking if this is actually creating an `omit_glenalfgknsfaklnef` volume.

File: ansible/roles/loadbalancer/templates/haproxy/haproxy_run.sh.j2 [hunk 3]
```diff
@@ -1,9 +1,40 @@
 #!/bin/bash -x
 
-# We need to run haproxy with one `-f` for each service, because including an
-# entire config directory was not a feature until version 1.7 of HAProxy.
-# So, append "-f $cfg" to the haproxy command for each service file.
-# This will run haproxy_cmd *exactly once*.
+{% if kolla_enable_tls_internal | bool or kolla_enable_tls_external | bool %}
+{% if kolla_enable_tls_external | bool %}
+if [ ! -e "/etc/haproxy/certificates/haproxy.pem" ]; then
```
Comment (markgoddard) [inline ansible/roles/loadbalancer/templates/haproxy/haproxy_run.sh.j2:5]: Would feel more comfortable if this was wrapped in an `if enable_letsencrypt`...

File: ansible/group_vars/all.yml [hunk 4]
```diff
@@ -942,6 +947,8 @@ rabbitmq_cacert: "/etc/ssl/certs/{{ 'ca-certificates.crt' if kolla_base_distro i
 ####################
 haproxy_user: "openstack"
 haproxy_enable_external_vip: "{{ 'no' if kolla_same_external_internal_vip | bool else 'yes' }}"
+haproxy_enable_http2: "yes"
```
[the remark this answers is not in the document]
Comment (mnasiadka) [reply ansible/group_vars/all.yml:950]: I assume most of users use http/2 compliant browsers - https://caniuse.com/http2

File: ansible/roles/loadbalancer/templates/haproxy/haproxy_run.sh.j2 (hunk 3 above)
Comment (mnasiadka) [reply ansible/roles/loadbalancer/templates/haproxy/haproxy_run.sh.j2:5]: https://review.opendev.org/c/openstack/kolla-ansible/+/904711

Comment (mnasiadka) [reply ansible/roles/loadbalancer/templates/haproxy/haproxy_run.sh.j2:5]: Although I wonder how much of a corner case it is, when you have tis enabled, but don't have a cert in the container...

File: ansible/roles/loadbalancer/defaults/main.yml (hunk 2 above)
Comment (mnasiadka) [reply ansible/roles/loadbalancer/defaults/main.yml:143]: Checked, it doesn't.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**